streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskExecutionMetadata.java

```diff
@@ -63,16 +68,52 @@ public ProcessingMode processingMode() {
 
     public boolean canProcessTask(final Task task, final long now) {
         final String topologyName = task.id().topologyName();
+        final boolean taskWasReady = !taskToLastNotReadyLogTime.containsKey(task.id());
+        final boolean canProcess;
+        final String logMessage;
+        
         if (!hasNamedTopologies) {
             // TODO implement error handling/backoff for non-named topologies (needs KIP)
-            return !pausedTopologies.contains(UNNAMED_TOPOLOGY);
+            logMessage = String.format("Task %s processing check for unnamed topology '%s'", task.id(), topologyName);
+            canProcess = !pausedTopologies.contains(UNNAMED_TOPOLOGY);
```

Comment:
Wondering if we should do this line first, and compute the log message based on the output of `canProcess` ?